Library/Homebrew/cmd/shellenv.sh

```diff
@@ -50,6 +52,7 @@ homebrew-shellenv() {
       echo "export PATH=\"${HOMEBREW_PREFIX}/bin:${HOMEBREW_PREFIX}/sbin\${PATH+:\$PATH}\";"
       echo "[ -z \"\${MANPATH-}\" ] || export MANPATH=\":\${MANPATH#:}\";"
       echo "export INFOPATH=\"${HOMEBREW_PREFIX}/share/info:\${INFOPATH:-}\";"
+      echo "[ -z \"\${XDG_DATA_DIRS-}\" ] || export XDG_DATA_DIRS=\"${HOMEBREW_PREFIX}/share:\${XDG_DATA_DIRS#:}\";"
```

Comment:
Actually, sorry, changed my mind: feels like this should probably be scoped to be Linux-only?

Reply:
Many formulae check `XDG_*` directories on both macOS and Linux. This is needed on macOS too to avoid Homebrew/homebrew-core#179217.

Reply:
Yeah, this is a cross-platform issue. It just isn't as visible on macOS because it doesn't set `XDG_DATA_DIRS` by default, but if a user _does_ set it, we're back to the same problem.

Reply:
> This is needed on macOS too to avoid [Homebrew/homebrew-core#179217](https://github.com/Homebrew/homebrew-core/issues/179217).

I don't see any mention of macOS in that issue?

I disagree with adding a new variable like this that's only widely used on Linux to all macOS users who run `brew shellenv`. `brew shellenv` itself could avoid outputting anything here if `XDG_DATA_DIRS` is unset which seems more sensible.

> It just isn't as visible on macOS because it doesn't set `XDG_DATA_DIRS` by default, but if a user _does_ set it, we're back to the same problem.

This doesn't feel like something that means we need this variable on every `brew shellenv` output for every macOS user.

Reply:
Given that most users just do something like
```sh
eval "$($HOMEBREW_PREFIX/bin/brew shellenv)"
```
does the exact content of `brew shellenv` really matter?

To put it another way:

If we adjust the code so that `brew shellenv` only spits out the lines involving `XDG_DATA_DIRS` if `XDG_DATA_DIRS` is set, then it only changes the timing of when the check for `XDG_DATA_DIRS` is done (and, arguably, doesn't change anything important).

In this PR, `XDG_DATA_DIRS` is checked by the user's shell after `brew shellenv` is run. If `brew shellenv` spits out the `XDG_DATA_DIRS` lines only if it's set, then `brew` will still check `XDG_DATA_DIRS`, except it does this before it prints anything.

Reply:
> does the exact content of `brew shellenv` really matter?

To me, yes, particularly given the temptation will just be to add more and more strange variables if we start adding any `XDG_*` ones.

There's other places we use `XDG_*` variables only on Linux. We don't even try to support them in Homebrew fully on macOS.

> (and, arguably, doesn't change anything important).

If it doesn't change anything important: let's make the change 😁 

> then `brew` will still check `XDG_DATA_DIRS`, except it does this before it prints anything.

This is fine to me because it slims the output of the command and sets a precedent for future changes like this.

> This is needed on macOS too to avoid [Homebrew/homebrew-core#179217](https://github.com/Homebrew/homebrew-core/issues/179217).

I stand by that: we've not seen any indication that any users have or will hit this on macOS and it's speculation to state that it's a cross-platform issue.

Reply:
> To me, yes, particularly given the temptation will just be to add more and more strange variables if we start adding any `XDG_*` ones.

The XDG Base Directory Specification is a well-known standard. It's very unlikely that adding them here will be used as a basis for adding more and more strange variables.

In any case, that's a slippery-slope fallacy. If we don't want to add more and more strange variables: let's not do that.

> There's other places we use `XDG_*` variables only on Linux. We don't even try to support them in Homebrew fully on macOS.

And I think that's a mistake.

> > This is needed on macOS too to avoid [Homebrew/homebrew-core#179217](https://github.com/Homebrew/homebrew-core/issues/179217).
> 
> I stand by that: we've not seen any indication that any users have or will hit this on macOS and it's speculation to state that it's a cross-platform issue.

I don't think it's speculation, and I don't think @gromgit does either. It seems weird to me that you think we're both just speculating.

The fact that it's a cross-platform issue is explained here: https://github.com/Homebrew/homebrew-core/issues/179217#issuecomment-2263273707

`glib` treats `XDG_DATA_DIRS` the same across macOS and Linux, and the issue seems to be from `lxi-tools` not being able to find its schema files because the user's `XDG_DATA_DIRS` was pointing at a location that didn't include Homebrew's prefix.

Reply:
> The XDG Base Directory Specification is a well-known standard.

It's a standard on Linux systems that Homebrew has a history of pushing back being used on macOS (hence my continued attempt to do so here). 

> In any case, that's a slippery-slope fallacy. If we don't want to add more and more strange variables: let's not do that.

I do not want to add this variable. It's not fallacious to suggest that adding this `XDG_*` variable makes it harder in future to argue against adding more.

> It seems weird to me that you think we're both just speculating.
> the user's `XDG_DATA_DIRS` was pointing at a location that didn't include Homebrew's prefix.

If a user reports this issue on macOS: let's consider a fix for macOS. I do not see evidence that users are setting this variable and having this issue on macOS. 

On "speculating" here I'm specifying referring to something Max Howell used to say about "speculatively fixing issues that you think could happen instead of fixing issues that are effecting users today". This issue occurring to our macOS users is not something I have evidence of yet. If you have a link of a report of this issue on macOS: please share it and I will revoke this claim.